archinstall/lib/disk/device_model.py

```diff
@@ -1204,10 +1204,21 @@ def __post_init__(self):
 			raise ValueError('LuksOnLvm encryption require LMV volumes to be defined')
 
 	def should_generate_encryption_file(self, dev: PartitionModification | LvmVolume) -> bool:
-		if isinstance(dev, PartitionModification):
+		warn("Deprecation warning: should_generate_encryption_file() has been renamed to should_generate_encryption_keyfile() and should_generate_encryption_file will be removed.")
+		return self.should_generate_encryption_keyfile(dev)
+
+	def should_generate_encryption_keyfile(self, dev: PartitionModification | LvmVolume) -> bool:
+		# If all partitions being encrypted, are non-root file systems
+		# we will not generate key-files as their safety can not be guaranteed.
+		if all([part.mountpoint != Path('/') for part in self.partitions]):
```

Comment:
just FYI there's a `part.is_root()` :)

Reply:
This check doesn't feel right here...on the caller side this is run for each partition now, but we can probably short circuit it 

```
	def generate_keyfile_for_dev(self, dev: PartitionModification | LvmVolume) -> bool:
		# If all partitions being encrypted, are non-root file systems
		# we will not generate key-files as their safety can not be guaranteed.
		if isinstance(dev, PartitionModification):
			return dev in self.partitions and dev.mountpoint != Path('/')
		elif isinstance(dev, LvmVolume):
			return dev in self.lvm_volumes and dev.mountpoint != Path('/')

		return False

	def generate_keyfiles(self) -> bool:
		# If all partitions being encrypted, are non-root file systems
		# we will not generate key-files as their safety can not be guaranteed.
		if all([part.mountpoint != Path('/') for part in self.partitions]):
			return True
		return False
```

And on the caller

```
	def _setup_luks_unlock_partitions(self) -> None:
		if not self._disk_encryption.generate_keyfiles():
			return
		...
```

What do you think?

Reply:
> just FYI there's a `part.is_root()` :)

Silly me just copied the code from two rows down, but now that you mention it I do remember seeing that function, I'll change!